system/include/platform.h

```diff
@@ -508,4 +508,22 @@ typedef __int64 cycle_t;
 #endif
 #endif
 
+//Use DECL_EXCEPTION for objects which are thrown if they are not exported from a dll
+#ifdef _MSC_VER
+ #define DECL_EXPORT __declspec(dllexport)
+ #define DECL_IMPORT __declspec(dllimport)
+ #define DECL_LOCAL
+ #define DECL_EXCEPTION
+#elif __GNUC__ >= 4
```

Comment:
Should we also make the use of -fvisibility=hidden in the makefile dependent on gnuc v4 ?